tests/unit_tests/commands/databases/columnar_reader_test.py

```diff
@@ -0,0 +1,253 @@
+    "Age": [30, 25, 20],
+    "City": ["city1", "city2", "city3"],
+    "Birth": ["1990-02-01", "1995-02-01", "2000-02-01"],
+}
+
+COLUMNAR_WITH_NULLS: dict[str, list[Any]] = {
+    "Name": ["name1", "name2", "name3"],
+    "Age": [None, 25, 20],
+    "City": ["city1", None, "city3"],
+    "Birth": ["1990-02-01", "1995-02-01", "2000-02-01"],
+}
+
+
+COLUMNAR_WITH_FLOATS: dict[str, list[Any]] = {
+    "Name": ["name1", "name2", "name3"],
+    "Age": [30.1, 25.1, 20.1],
+    "City": ["city1", "city2", "city3"],
+    "Birth": ["1990-02-01", "1995-02-01", "2000-02-01"],
+}
+
+
+@pytest.mark.parametrize(
+    "file, options, expected_cols, expected_values",
+    [
+        (
+            create_columnar_file(COLUMNAR_DATA),
+            ColumnarReaderOptions(),
+            ["Name", "Age", "City", "Birth"],
+            [
+                ["name1", 30, "city1", "1990-02-01"],
+                ["name2", 25, "city2", "1995-02-01"],
+                ["name3", 20, "city3", "2000-02-01"],
+            ],
+        ),
+        (
+            create_columnar_file(COLUMNAR_DATA),
+            ColumnarReaderOptions(
+                columns_read=["Name", "Age"],
+            ),
+            ["Name", "Age"],
+            [
+                ["name1", 30],
+                ["name2", 25],
+                ["name3", 20],
+            ],
+        ),
+        (
+            create_columnar_file(COLUMNAR_DATA),
+            ColumnarReaderOptions(
+                columns_read=[],
+            ),
+            ["Name", "Age", "City", "Birth"],
+            [
+                ["name1", 30, "city1", "1990-02-01"],
+                ["name2", 25, "city2", "1995-02-01"],
+                ["name3", 20, "city3", "2000-02-01"],
+            ],
+        ),
+        (
+            create_columnar_file(COLUMNAR_WITH_NULLS),
+            ColumnarReaderOptions(),
+            ["Name", "Age", "City", "Birth"],
+            [
+                ["name1", np.nan, "city1", "1990-02-01"],
+                ["name2", 25, None, "1995-02-01"],
+                ["name3", 20, "city3", "2000-02-01"],
+            ],
+        ),
+        (
+            create_columnar_file(COLUMNAR_WITH_FLOATS),
+            ColumnarReaderOptions(),
+            ["Name", "Age", "City", "Birth"],
+            [
+                ["name1", 30.1, "city1", "1990-02-01"],
+                ["name2", 25.1, "city2", "1995-02-01"],
+                ["name3", 20.1, "city3", "2000-02-01"],
+            ],
+        ),
+    ],
+)
+def test_columnar_reader_file_to_dataframe(
+    file, options, expected_cols, expected_values
+):
+    reader = ColumnarReader(
+        options=options,
+    )
+    df = reader.file_to_dataframe(file)
+    assert df.columns.tolist() == expected_cols
+    actual_values = df.values.tolist()
+    for i in range(len(expected_values)):
+        for j in range(len(expected_values[i])):
+            expected_val = expected_values[i][j]
+            actual_val = actual_values[i][j]
```

Comment:
Could simplify to:

```suggestion
    for expected_row, actual_row in zip(expected_values, actual_values):
        for expected_val, actual_val in zip(expected_row, actual_row):
```

Though my suggestion and your original code could both fail silently if `actual_values` has more rows or more columns than `expected_values`.